R/obsPlot.R

```diff
@@ -1,274 +1,453 @@
-#' @param boxNotchWidth Set the width of box notches (0-1) (Default = 0.8)
-#' @param pointSize Size of the points (Default = 4)
-#' @param pointJitter Amount to jitter the points (Default = 0) Try 0.2 if you
-#'   have a lot of points.
-#' @param meanColor Color for the mean layer (Default = "red2")
-#' @param meanFill Fill color for the mean layer (Default = "goldenrod1")
-#' @param meanShape Shape for the mean layer (Default = 21; fillable circle)
-#' @param meanAlpha Transparency for the mean layer (Default = 0.7)
-#' @param meanSize Size of the mean points (Default = 3)
-#' @param baseFontSize The smallest size font in the figure in points. (Default =
-#'   12)
-#' @param themeStyle "bw" or "grey" which correspond to theme_bw or theme_grey
-#'   respectively. Default = bw"
+#' @param color Color for the boxplot or the violin ploy (default = "deepskyblue3")
 #' @param facet Specifies whether to facet (TRUE) or print individual plots
-#'   (FALSE)  (Default = TRUE)
-#' @param facetRow Explicitly set the number of rows for the facet plot.
-#'   Default behavior will automatically set the columns. (Default = NULL)
-#' @param xAngle Angle to set the sample labels on the X axis. (Default =  30; Range = 0-90)
-#' @param scales Specify same scales or independent scales for each subplot (Default = "free_y";
-#'   Allowed values: "fixed", "free_x", "free_y", "free")
+#'   (FALSE)  (default = TRUE)
+#' @param axisFree Specify same scale or independent scales for each subplot (default = TRUE;
+#'   Allowed values: TRUE and FALSE)
 #'
-#' @return ggplot. If Facet = TRUE (default) returns a faceted ggplot object. If
-#'   facet = FALSE, returns a list of ggplot objects indexed
+#' @return Plot of type canvasXpress or ggplot. If Facet = TRUE (default) returns a faceted object. If
+#'   facet = FALSE, returns a list of objects indexed
 #'   by observation (gene) names.
 #'
 #' @examples
 #' \dontrun{
-#'   # Simulate some data with row and colnames
-#'   groups <- paste("group", factor(rep(1:4, each = 100)), sep = "")
-#'   x <- matrix(rnorm(2400, mean = 10), ncol = length(groups))
-#'   colnames(x) <- paste("sample", 1:ncol(x), sep = "")
-#'   rownames(x) <- paste("gene", 1:nrow(x), sep="")
-#'   # Reformat into tidy dataframe
-#'   tidyInt <- tidyIntensity(x)
-#'
-#'  # Or get data from a DGEobj with RNA-Seq data
-#'  log2cpm <- convertCounts(dgeObj$counts, unit = "cpm", log = TRUE, normalize = "tmm")
-#'  tidyInt <- tidyIntensity(log2cpm,
-#'                           rowidColname = "GeneID",
-#'                           keyColname = "Sample",
-#'                           valueColname = "Log2CPM",
-#'                           group = dgeObj$design$ReplicateGroup)
 #'
 #'   # Faceted boxplot
-#'   obsPlot(tidyInt,
-#'            plotByCol = "GeneID",
-#'            groupCol = "group",
-#'            valueCol ="Log2CPM",
-#'            pointJitter = 0.1,
-#'            facetRow = 2)
+#'   obsPlot(DGEobj,
```

Comment:
That is Connie's requirement. That we do not stop the user from plotting in case of large number of facets but just throw a warning. They will need to subset the DGEobj to reduce the number of Facets. Message was given Connie. I will need to check with her if I have to update the message to indicate how to reduce the number of facets in the plot.